File: 2105.cpp
struct Solution {
    inline int minimumRefill(vector<int>& plants, int capacityA, int capacityB) {
        int n = plants.size(), wa = capacityA, wb = capacityB, refill = 0;
        int a = 0, b = n - 1;
        while (a < b){
            if (plants[a] > wa){
                ++refill;
                wa = capacityA;
            }
            wa -= plants[a++];
            if (plants[b] > wb){
                ++refill;
                wb = capacityB;
            }
            wb -= plants[b--];
        }
        if (a == b){
            refill += (plants[a] > max(wa, wb));
        }
        return refill;
    }
};

File: 2105.py
class Solution:
    def minimumRefill(self, plants: List[int], capacityA: int, capacityB: int) -> int:
        n = len(plants)
        a = 0
        b = n - 1
        wa = capacityA
        wb = capacityB
        refill = 0
        while a < b:
            if plants[a] > wa:
                refill += 1
                wa = capacityA
            wa -= plants[a]
            a += 1
            if plants[b] > wb:
                refill += 1
                wb = capacityB
            wb -= plants[b]
            b -= 1
        if a == b:
            refill += (plants[a] > max(wa, wb))
        return refill
